### PR TITLE
Add Hyperlink Support to the Version Modal

### LINKED ISSUE #426: Add Hyperlink Support
For now the repository url is just a text, so add hyperlink support to open the url in the browser.

### PR DESCRIPTION
**Pull request by sundaram123krishnan**
Closes #426

### CODE REVIEW

File: src/views/modal_handler.rs [hunk 1]
```diff
@@ -59,11 +64,19 @@ pub(crate) const MODAL_CONTAINER_STYLE: ContainerStyle = ContainerStyle {
...
+    border_radius: 2.0,
+    hovered_bg_color: Color::TRANSPARENT,
+    hovered_text_color: Color::from_rgba(0.0, 0.0, 0.6, 1.0),
+};
 
 impl DisplayModal {
     pub fn new() -> Self {
         Self {
             show_modal: false,
+            is_warning: false,
```
Comment (andrewdavidmackenzie) [inline src/views/modal_handler.rs:79]: Mor ebooleans creeping in here.

I suggest you have an enum (`DialogType`) that has variants for each of the dialogs to show (HardwareDetails, Version, Exit..) (containing information specific to each one if that is applicable), and then have a message ShowDialog(DialogType) that is sent.

If needed, you can store the current dialog being shown in a self.dialog_type : DialogType field (but not sure if needed?).

File: src/views/modal_handler.rs [hunk 2]
```diff
@@ -82,6 +95,7 @@ impl DisplayModal {
             // Display warning for unsaved changes
             ModalMessage::UnsavedChangesExitModal => {
                 self.show_modal = true;
+                self.is_warning = true;
                 self.modal_type = Some(ModalType::Warning {
```
Comment (andrewdavidmackenzie) [inline src/views/modal_handler.rs:99]: Aha, you have very similar to what I was saying - can we use that and avoid the boolean flags for show_modal, and is_warnign and is_version?

File: src/views/modal_handler.rs [hunk 3]
```diff
@@ -120,13 +136,22 @@ impl DisplayModal {
...
+                self.is_warning = false;
                 self.modal_type = Some(ModalType::Info {
                     title: "About Piggui".to_string(),
                     body: crate::views::version::version().to_string(),
+                    is_version: true,
                 });
                 Command::none()
             }
 
+            ModalMessage::OpenRepoLink => {
```
Comment (andrewdavidmackenzie) [inline src/views/modal_handler.rs:148]: It's pretty poor that Iced (or Iced_aw) doesn't let you put hypertext in a text in a dialog and handle the links for you, and we have to do all this ... :-( 

Consider a PR to them to improve this as a personal contribution?

File: src/views/modal_handler.rs [hunk 4]
```diff
@@ -198,16 +223,38 @@ impl DisplayModal {
...
-                        .on_press(Message::ModalHandle(ModalMessage::HideModal))
-                        .style(MODAL_CANCEL_BUTTON_STYLE.get_button_style()),
-                );
+                let mut hyperlink_row = Row::new().width(Length::Fill);
+                let mut button_row = Row::new();
+                if *is_version {
+                    hyperlink_row = hyperlink_row.push(Text::new("Full source available at: "));
+                    hyperlink_row = hyperlink_row
+                        .push(
+                            button(Text::new("Project Repository"))
```
Comment (andrewdavidmackenzie) [inline src/views/modal_handler.rs:240]: I'd put the link here, so the user knows where you are about to send them.
Either "github" or the full url "https://github...."?

File: src/views/modal_handler.rs [hunk 5]
```diff
@@ -198,16 +223,38 @@ impl DisplayModal {
...
+                                .on_press(Message::ModalHandle(ModalMessage::OpenRepoLink))
+                                .style(HYPERLINK_BUTTON_STYLE.get_button_style()),
```
Comment (andrewdavidmackenzie) [inline src/views/modal_handler.rs:242]: What about the message (OpenReploLink) be "OpenURL" and have the URL inside it (OpenUrl("https://github...")), then you can re-use it for any Url in any text in the dialog?

File: src/views/version.rs [hunk 6]
```diff
@@ -7,7 +7,7 @@ const VERSION: &str = env!("CARGO_PKG_VERSION");
 const BIN_NAME: &str = env!("CARGO_BIN_NAME");
 const PKG_NAME: &str = env!("CARGO_PKG_NAME");
 const LICENSE: &str = env!("CARGO_PKG_LICENSE");
-const REPOSITORY: &str = env!("CARGO_PKG_REPOSITORY");
+pub const REPOSITORY: &str = env!("CARGO_PKG_REPOSITORY");
```
Comment (andrewdavidmackenzie) [inline src/views/version.rs:10]: +1

**Review by andrewdavidmackenzie — COMMENTED** [review]
I think that some small changes should be made as part of this PR. 
If you want to then continue to the bigger changes in the PR, then OK.
But if not, then please write an issue to continue to work on it in another PR ?
- reduce use of boolean flags and use enums and the variant's contents
- reusable OpenURL message
-

**Comment by sundaram123krishnan** [comment]
Opened issue regarding the refactor, https://github.com/andrewdavidmackenzie/pigg/issues/437, thinking to address all the above changes in that PR.

File: src/views/version.rs (hunk 6 above)
Comment (andrewdavidmackenzie) [reply src/views/version.rs:10]: Needs top be pub?

**Pull request merged**